arch/arc/CMakeLists.txt

```diff
@@ -20,3 +20,28 @@ if(CONFIG_ISA_ARCV2)
 endif()
 
 add_subdirectory(core)
+
+
+
+if (CONFIG_BUILD_WITH_SECURESHIELD)
+  set(SECURESHIELD_BINARY_DIR ${CMAKE_BINARY_DIR}/secureshield)
+  file(MAKE_DIRECTORY ${SECURESHIELD_BINARY_DIR})
+  add_custom_target(secureshield_cmake
+    COMMAND west build -p auto -b ${CONFIG_SECURESHIELD_BOARD} -d ${SECURESHIELD_BINARY_DIR} ${ZEPHYR_BASE}/${CONFIG_SECURESHIELD_IMAGE_PATH}
+    WORKING_DIRECTORY ${SECURESHIELD_BINARY_DIR}
+    COMMAND_EXPAND_LISTS
+  )
```

Comment:
I'm really not sure about this. I think @tejlmand should confirm this is OK. I know we do some things with sub-images in TF-M (so cc @ioannisg and @microbuilder for help here) but I'm not sure if this style of calling out to west build from the build system itself is something we should encourage.

@nashif -- any feedback?

Reply:
Marti, can you explain your concern? The goal here is to make it easy for users to combine building/flashing/debugging a secure and a normal binary. In the TF-M case, the secure binary is build as an external module I believe. For ARC we decided to have a more closer Zephyr integration and use west to build the secure binary.
Do you have a suggestion for a different (better?) way of achieving the same (we want to e.g. be able to leverage twister and CI etc. instead of current manual steps for combining the secure and normal images)?

Reply:
@ruuddw the topic of building multiple images within the zephyr build system has a long and contentious history. I won't go into all the details in this comment -- there was even an entire BoF talk about it at ELCE a few years ago to discuss a proposal from Nordic on it which was ultimately rejected by the Zephyr community: https://www.youtube.com/watch?v=ZZJ-r94MUnI

Given that history I think it's worth making sure everyone knows about and is in agreement with the approach here.

Reply:
@mbolivar, thanks for the pointer/reminder. Unfortunately the discussion is hard to follow in the recording.
The approach here isn't following the 'nonrecursive cmake' which was challenged to introduce too much complexity to handle the namespaces. It follows the 'recursive cmake' - we'll pay extra attention to handling dependencies, but I think we'll be fine with only limited dependencies in this case.